### ISSUE TITLE
Shrinking the Virtuoso List to fit a shorter list of contents

### ISSUE THREAD

**Issue by nelson-liu** [petyosi/react-virtuoso#652]
**Describe the bug**

Hi! Virtuoso works great for the (intended) use-case of displaying thousands of items, thanks for building it. I'm using a virtuoso list in a context where sometimes I need to make a list of thousands of items, and sometimes i need to make a list of only a few items (let's say 2 items in this example). This is being used in the context of a drop-down list.

In this latter setting, where I only have a few items to render, I'm having difficulty making Virtuoso shrink to fit the items. There's a large white space where virtuoso would normally render more items, but there aren't any such items to render.

**Reproduction**
Here's a codesandbox that demonstrates the issue: https://codesandbox.io/s/sandpack-project-forked-evi14k?file=/App.js .

In this case, a Virtuoso list is rendering 2 items. I set the backgroundcolor of virtuoso to be red and added a dotted border so it's visually apparent that the Virtuoso scroller is much larger than list of elements it has to render (hence why the red background shows up, and why the dotted outline is far below the second item).

Is there any way to fix the parent divs or the Virtuoso list in the example above, such that Virtuoso shrinks to fit the contents (and thus we don't see the red background, and the dotted outline is tightly wrapped around the 2 rendered items)?

**Expected behavior**

I'd like the virtuoso list to shrink to fit the items. This is desirable in the case of a drop-down because it's pretty visually unappealing to have a large white space filling the unused scroller height.

**Screenshots**
What I see in the codesandbox:

![image](https://user-images.githubusercontent.com/7272031/166174640-8958e8c7-3a86-486b-a0d7-555c4e0f0cf6.png)

Thanks in advance, and thanks again for building such a useful library!

**Comment by nelson-liu** [comment]
I wonder if there's something clever / hacky you can do with the scrollerRef here. Given a scrollerRef, is it possible to get the height of the elements on the inside? (sorry if this suggestion is way off, at my wits end here :) )

**Comment by nelson-liu** [comment]
Update: I was able to do this by using the `totalListHeightChanged` prop. In particular, I added a new piece of state to store the list height (`let's just call it self.state.containerHeight`), and used totalListHeightChanged to set this state (just setting the state to the new height).

Then, in the div wrapping the `Virtuoso` instance, I set the height with the style prop as `this.state.containerHeight`, and set `maxHeight` as well. Not sure if this is the approach you would have recommended, but it seems to be working ok. Happy to hear suggestions if you have anything better in mind, though!
